source/slang/slang-emit-source-writer.cpp

```diff
@@ -251,7 +252,13 @@ void SourceWriter::emit(double value)
     std::ostringstream stream;
     stream.imbue(std::locale::classic());
     stream.setf(std::ios::fixed, std::ios::floatfield);
-    stream.precision(20);
+
```

Comment:
The default behavior of ostringstream is what you described.
Because we set it to be "fixed" at the line above, the output value is always in the fixed format and never in the scientific format.
I assumed that we are using "fixed" format for a reason but I guess we can remove it judging by your comment.

I will make some changes in a way not to use "std::ios::fixed" flag.

Reply:
The default setting won't work either because it doesn't emit enough digits when in scientific notation. We still need at least 20 digits when in scientific notation.